src/Integration.Vsix/Suppression/SuppressionManager.cs

```diff
@@ -0,0 +1,128 @@
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Suppression
+{
+    internal sealed class SuppressionManager : IDisposable
+    {
+        private readonly IServiceProvider serviceProvider;
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+
+        private DelegateInjector delegateInjector;
+        private LiveIssueFactory liveIssueFactory;
+        private ISonarQubeIssuesProvider sonarqubeIssueProvider;
+
+        public SuppressionManager(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+            activeSolutionBoundTracker = serviceProvider.GetMefService<IActiveSolutionBoundTracker>();
+            activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
+
+            RefreshSuppresionHandling();
+        }
+
+        private void RefreshSuppresionHandling()
+        {
+            if (activeSolutionBoundTracker.IsActiveSolutionBound)
+            {
+                SetupSuppressionHandling();
+            }
+            else
+            {
+                CleanupSuppressionHandling();
+            }
+        }
+
+        private void SetupSuppressionHandling()
+        {
+            liveIssueFactory = new LiveIssueFactory(serviceProvider);
+            delegateInjector = new DelegateInjector(ShouldIssueBeSuppressed, serviceProvider);
+            sonarqubeIssueProvider = this.serviceProvider.GetMefService<ISonarQubeIssuesProvider>(); // Cannot do new because of IHost
+        }
+
+        private void CleanupSuppressionHandling()
+        {
+            delegateInjector?.Dispose();
+            delegateInjector = null;
+            liveIssueFactory = null;
+            (sonarqubeIssueProvider as IDisposable)?.Dispose();
+            sonarqubeIssueProvider = null;
+        }
+
+        private void OnSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs e)
+        {
+            RefreshSuppresionHandling();
+        }
+
+        private bool ShouldIssueBeSuppressed(Diagnostic diagnostic)
+        {
+            // This method is called for every analyzer issue that is raised so it should be fast.
+            if (!diagnostic.Location.IsInSource) { return false; }
+            if (activeSolutionBoundTracker == null || !activeSolutionBoundTracker.IsActiveSolutionBound) { return false; }
+
+            LiveIssue liveIssue = liveIssueFactory.Create(diagnostic);
+            if (liveIssue == null)
+            {
+                return false; // Unable to get the data required to map a Roslyn issue to a SonarQube issue
+            }
+
+            // Issues match if:
+            // 1. Same component, same file, same error code, same line hash        // tolerant to line number changing
+            // 2. Same component, same file, same error code, same line             // tolarant to code on the line changing e.g. var rename
+
+            // TODO: ?need to make file path relative to the project file path
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)